### PR TITLE
default_snat_status attribute added

### PR DESCRIPTION
**Pull request by venkykuberan**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Cluster: added default_snat_status for `google_container_cluster` resource

```
reference: https://github.com/terraform-providers/terraform-provider-google/issues/6465

### CODE REVIEW

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -3170,6 +3234,16 @@ func flattenClusterTelemetry(c *containerBeta.ClusterTelemetry) []map[string]int
 	return result
 }
 
+func flattenDefaultSnatStatus(c *containerBeta.DefaultSnatStatus) []map[string]interface{} {
+	result := []map[string]interface{}{}
+	if c != nil {
```
Comment (ndmckinley) [inline third_party/terraform/resources/resource_container_cluster.go.erb:3239]: You may want to return disabled: true if c == nil, it helps `plan` make better decisions.  :)

Comment (venkykuberan) [reply third_party/terraform/resources/resource_container_cluster.go.erb:3239]: that makes the default value as `disabled: true`, right ?. What if i don't want to set the attribute if its not in the config ?

Comment (venkykuberan) [reply third_party/terraform/resources/resource_container_cluster.go.erb:3239]: Added tests for this attribute.

**Comment by danawillow** [comment]
I think this PR is causing VCR test failures. @venkykuberan , can you double check? The error is:
```
------- Stdout: -------
=== RUN   TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock
=== PAUSE TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock
=== CONT  TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock
TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock: testing.go:667: Step 0, expected error:
config is invalid: Unsupported block type: Blocks of type "default_snat_status" are not expected here.
To match:
master_ipv4_cidr_block must be set if enable_private_nodes == true
--- FAIL: TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock (0.33s)
FAIL
```

and since this is a VCR test failure, it's happening in the GA provider.

**Comment by ndmckinley** [comment]
Ah, shoot, I looked here: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstreamVcr/134556? to make sure it was passing!  :(

**Pull request merged**